### DISCUSSION TITLE
access dags from multiple folders

### DISCUSSION THREAD

**Discussion by t4n1o** [apache/airflow#18906] [Q&A]
Is there a way to access dags from multiple folders?

## Context:
I have dags & scripts to run in 2-3 git repositories. I push these repos to the server in an automated manner. I just want to use symlinks or dagbag to let airflow parse more than 1 folder for dags.

The alternative is to make all these different repos copy their dags to the dag folder with an automated bash script.
____
**What are the best practicies for multiple dag folders? Copy them into the main one? Use symlinks?**

**Comment by potiuk** [comment]
You can use an umbrella "git" repository which will add those different git repos as submodules in subfloders and use that repository to checkout in "DAG" folder. GitSync has an option to enable submodule update. This has the advantage, that the checkout will be automatically linked to the specific commit hashes in the submodules, and that the git sync will be atomic (this is how GitSync works - it atomically swaps folders when whole checkout is complete). You can then control which version of which submodule is used in the "umbrella" by updating submodules to specific (or latest) versions of the submodules - either manually or automatically. 

No need to copy stuff around.
